File: frontend/src/app/models/order.ts
import { ICartProduct } from "./product";

export type TypeOrder = {
    created: string,
    status: string,
    order_products: ICartProduct[],
}

export type TypeCreateOrder = {
    name: string | null | undefined,
    secondname: string | null | undefined,
    delivery: string | null | undefined,
    payment: string | null | undefined,
    adress: string | null | undefined,
    phone: string | null | undefined,
}

File: backend/app/domain/order/order.py
from dataclasses import dataclass
from datetime import datetime

from domain.product.product import BaseProduct


@dataclass
class OrderProduct(BaseProduct):
    quantity: int
    size: str
    image: str


@dataclass
class Order:
    name: str
    secondname: str
    adress: str
    phone: str
    payment: str
    delivery: str
    user_id: int
    created: datetime | str
    order_products: list[OrderProduct]
